core/src/main/scala/org/apache/spark/status/api/v1/OneApplicationResource.scala

```diff
@@ -51,6 +51,52 @@ private[v1] class AbstractApplicationResource extends BaseAppResource {
   @Path("executors")
   def executorList(): Seq[ExecutorSummary] = withUI(_.store.executorList(true))
 
+  @GET
+  @Path("executors/{executorId}/threads")
+  def threadDump(@PathParam("executorId") executorId: String): Response = withUI { ui =>
```

Comment:
If you throw exceptions in the code below instead, you can have a more specialized return type here, which I slightly prefer. 

Not all errors currently have exceptions (e.g. 503 is missing), but it's easy to add them.

(I also don't remember exactly what version of JAX-RS is being used by Spark, but if it's 2.1, that version has all these exceptions already, so some cleanup could be done here. 2.0 doesn't have them though.)

Reply:
Review comments are applied then retested and test results are updated.
Moreover I have removed the condition with UIUtils.stripXSS as the existing check (whether the executorId is  'driver' or number) seams to me enough. 

Some if conditions on options are transformed to a val initialised with a getOrElse as the option value is safe to be used elsewhere. 

As I have seen the JAX RS is at 2.0.1, so I did not touch the mapping of exceptions to JSON.